eq_cir_proxy_service/utils/iap.py

```diff
@@ -0,0 +1,71 @@
+"""Utility functions for handling IAP authentication and HTTP clients."""
+
+import os
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+from typing import cast
+
+import google.auth.transport.requests
+import google.oauth2.id_token
+from httpx import AsyncClient
+from structlog import get_logger
+
+logger = get_logger()
+
+
+def get_iap_token(audience: str) -> str:
+    """Fetch an ID token for the IAP-secured resource (blocking)."""
+    request = google.auth.transport.requests.Request()  # type: ignore[no-untyped-call]
+    token = google.oauth2.id_token.fetch_id_token(request, audience)  # type: ignore[no-untyped-call]
+    if token is None:
+        logger.error("Failed to fetch IAP token", audience=audience)
+        error_message = f"Failed to fetch IAP token for audience {audience}"
+        raise RuntimeError(error_message)
+    return cast(str, token)
+
+
+def unknown_env_error(env: str) -> str:
```

Comment:
This begs for a new error class defined by us (and customising the "___str___" magic method (string representations for objects), example from Runner:
Definition:
```python
class DataVersionError(Exception):
    def __init__(self, version: str):
        super().__init__()
        self.version = version

    def __str__(self) -> str:
        return f"Data version {self.version} not supported"
```
and later in the code:
```python
raise DataVersionError(schema.json["data_version"])
```